src/main/java/com/softserve/bookstore/repositories/BookRepository.java

```diff
@@ -3,39 +3,109 @@
 
 import com.softserve.bookstore.connection.ConnectionManager;
 import com.softserve.bookstore.data.ReadDataFromBookFile;
+import com.softserve.bookstore.data.utils.AuthorUtility;
+
 import com.softserve.bookstore.models.Author;
 import com.softserve.bookstore.models.Book;
+import com.softserve.bookstore.models.Genre;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
-
+import org.tinylog.Logger;
 import java.io.IOException;
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+
 
 @Repository
 public class BookRepository {
 
+    @Autowired
+    private AuthorRepository authorRepository;
```

Comment:
group public with public and private with private fields, also there is an order of these convention you can find on the internet
CodeStyle plugin might help figure out